core/src/main/java/org/elasticsearch/action/bulk/BulkRequest.java

```diff
@@ -300,10 +300,16 @@ public BulkRequest add(BytesReference data, @Nullable String defaultIndex, @Null
                 if (token == null) {
                     continue;
                 }
-                assert token == XContentParser.Token.START_OBJECT;
+                if(token != XContentParser.Token.START_OBJECT) {
+                    throw new IllegalArgumentException("Malformed action/metadata line [" + line + "], expected "
+                        + XContentParser.Token.START_OBJECT + " but found [" + token + "]");
+                }
                 // Move to FIELD_NAME, that's the action
                 token = parser.nextToken();
-                assert token == XContentParser.Token.FIELD_NAME;
+                if(token != XContentParser.Token.FIELD_NAME) {
```

Comment:
I think it would be good to use `XContentParserUtils#ensureExpectedToken` or `XContentParserUtils#ensureFieldName`